src/compounds/product-table/src/components/row.tsx

```diff
@@ -56,6 +55,23 @@ const ProductTableRow: React.FC<RowProps> = ({
 
   const cols = nonNullChildren.length
 
+  const [sizes, setSizes] = React.useState(() => {
+    const initialSizes: string[] = []
+    for (let i = 0; i < cols; i++) {
+      initialSizes.push('1fr')
+    }
+    return initialSizes
+  })
+
+  const setSize = (i: number, size: string) => {
+    // IF YOU BREAK THIS IF STATEMENT YOU WILL CREATE AN INFINITE LOOP
+    // @todo is there a better way of doing this? It runs n*2 times where n is
+    // every child with a customSize, and causes a rerender every time
+    if (size !== sizes[i]) {
+      setSizes([...sizes.slice(0, i), size, ...sizes.slice(i + 1)])
+    }
+  }
+
```

Comment:
This could be inefficient if there's a lot of cells with custom sizes - e.g. if there's four custom sizes, it will cause 16 rerenders.

After talking with barney, sounds like it would be better to read from `child.props.customSize` instead - or just define directly on row. Would mean we wouldn't be able to have default values like the one set in cell-image.tsx at the moment, so that would have to be dealt with in the product table builder.